lib/IDE/Utils.cpp

```diff
@@ -348,15 +348,6 @@ bool ide::initCompilerInvocation(
   // We don't care about LLVMArgs
   FrontendOpts.LLVMArgs.clear();
 
-  // SwiftSourceInfo files provide source location information for decls coming
-  // from loaded modules. For most IDE use cases it either has an undesirable
-  // impact on performance with no benefit (code completion), results in stale
-  // locations being used instead of more up-to-date indexer locations (cursor
-  // info), or has no observable effect (diagnostics, which are filtered to just
-  // those with a location in the primary file, and everything else).
-  if (shouldOptimizeForIDE)
```

Comment:
Can you go into more detail about the impact here on code-completion?  Is it now going to deserialize locations, or are we avoiding that in some other way?

Reply:
`swift::ide::CompletionInstance::performOperation` has `Invocation.getFrontendOptions().IgnoreSwiftSourceInfo = true;`, ie. it's hardcoded to ignore them regardless of OptimizeForIDE.